app/code/core/Mage/Catalog/Model/Resource/Product/Flat/Indexer.php

```diff
@@ -274,6 +272,10 @@ public function getAttribute($attributeCode)
                 ->getEntity();
             $attribute->setEntity($entity);
 
+            if (!($attribute instanceof Mage_Eav_Model_Entity_Attribute)) {
+                Mage::throwException('Product attribute(code=". $attributeCode . ") is expected to be of type Mage_Eav_Model_Entity_Attribute');
```

Comment:
Should just be this since it is not translated or meant for end users to see, also not properly quoted for variable expansion:
```suggestion
                throw new Exception('Product attribute(code=' . $attributeCode . ') is expected to be of type Mage_Eav_Model_Entity_Attribute');
```

Reply:
Will change, oversight. There is a second location where the same logic applies.